objstore.go

```diff
@@ -196,6 +196,26 @@ func ApplyIterOptions(options ...IterOption) IterParams {
 	return out
 }
 
+type UploadObjectParams struct {
+	ContentType string
+}
+
+type ObjectUploadOption func(f *UploadObjectParams)
```

Comment:
I think we can simplify (no need for params to be public struct):

```suggestion
type UploadOption func(f *uploadOptions)
```